react/src/pages/datasets/DatasetTable.tsx

```diff
@@ -0,0 +1,168 @@
+}
+
+export default function DatasetTable() {
+    const classes = useStyles();
+    const [showRunner, setRunner] = useState(false);
+    const [selectedDatasets, setSelectedDatasets] = useState<Dataset[]>([]);
+    const [datasetType, setDatasetType] = useState("");
+
+    // TODO: replace with enum values from backend
+    const tissueSampleTypes = toKVPair(["Blood", "Saliva", "Skin"]);
+    const datasetTypes = toKVPair(["WGS", "WES"]);
+    const conditions = toKVPair(["Control", "GermLine", "Somatic"]);
+
+    return (
+        <div>
+            <AnalysisRunnerDialog
+                datasets={selectedDatasets}
+                pipelines={pipelines}
+                open={showRunner}
+                onClose={() => setRunner(false)}
+            />
+            <MaterialTable
+                columns={[
+                    { title: 'Participant', field: 'participant_codename', editable: 'never' },
+                    { title: 'Family', field: 'family_codename', editable: 'never' },
+                    { title: 'Tissue Sample', field: 'tissue_sample_type', lookup: tissueSampleTypes },
+                    { title: 'Dataset Type', field: 'dataset_type', defaultFilter: datasetType, lookup: datasetTypes },
+                    { title: 'Condition', field: 'condition', lookup: conditions },
+                    { title: 'Notes', field: 'notes' },
+                    // { title: 'Created', field: 'created', type: 'datetime' },
+                    // { title: 'Created by', field: 'created_by' },
+                    { title: 'Updated', field: 'updated', type: 'datetime' },
+                    { title: 'Updated By', field: 'updated_by' },
+                ]}
+                data={rows}
+                title="Datasets"
+                options={{
+                    pageSize: 10,
+                    selection: true,
+                    filtering: true,
+                    search: false
+                }}
+                editable={{
+                    onRowUpdate: (newData, oldData) =>
+                        new Promise((resolve, reject) => {
+                            setTimeout(() => {
+                                // const dataUpdate = [...data];
+                                // const index = oldData.tableData.id;
+                                // dataUpdate[index] = newData;
+                                // setData([...dataUpdate]);
+
+                                resolve();
+                            }, 1000);
+                        }),
+                }}
+                components={{
+                    Toolbar: props => (
+                        <div>
+                            <MTableToolbar {...props} />
+                            <div className={classes.chipBar}>
+                                {[...new Set(rows.map(e => e.dataset_type))].map(type => (
+                                    <Chip label={type} onClick={() => setDatasetType(type)} clickable className={classes.chip} />
```

Comment:
When I clicked the chip button, the page goes blank. You can solve this problem by making `datasetType` an array and passing an array containing the filter values to `setDatasetType`.
```suggestion
                                    <Chip label={type} onClick={() => setDatasetType([type])} clickable className={classes.chip} />
```

Reply:
Great catch, thanks!